src/holder.ts

```diff
@@ -93,15 +99,23 @@ export class Holder {
       throw new SDJWTVCError('No holder public key in SD-JWT');
     }
 
-    const { keyBindingJWT } = await this.getKeyBindingJWT(options.audience, options.nonce);
+    let sdHashAlgorithm = jwt.payload['_sd_alg'] as string;
+    if (!sdHashAlgorithm) {
+      sdHashAlgorithm = defaultHashAlgorithm;
+    }
+
+    const hasher: Hasher = hasherCallbackFn(sdHashAlgorithm);
+    const sdJwtHash: string = hasher(sdJWT);
```

Comment:
> sd_hash: REQUIRED. The base64url-encoded hash digest over the Issuer-signed JWT and the selected Disclosures as defined below.

https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-07.html#section-5.3-3.2.1

Wouldn't this involve creating a hash of sdJWT and the selected disclosures?

```
const vcSDJWTWithRevealedDisclosures = this.revealDisclosures(sdJWT, disclosedList);
const sdJwtHash: string = hasher(vcSDJWTWithRevealedDisclosures);
```

Reply:
based on the draft-07 spec
then yes, it should be the following

```
const vcSDJWTWithRevealedDisclosures = this.revealDisclosures(sdJWT, disclosedList);
const sdJwtHash: string = hasher(vcSDJWTWithRevealedDisclosures);
```

@gmirc12 thanks for the PR, can you also make the above changes and hash the `vcSDJWTWithRevealedDisclosures` instead?


@vijayshiyani 
There's probably some confusion on which compactSDJWT to hash.

I think the example in the README.md for `presentVCSDJWT` could be improved

The `disclosedList` should be the following
and might be better named as `selectedDisclosures` or something similar to convey its purpose

 ```
   const selectedDisclosures = [
    {
      disclosure: 'WyJNcEFKRDhBWVBQaEJhT0tNIiwibmFtZSIsInRlc3QgcGVyc29uIl0'
    },
    {
      disclosure: 'WyJJbFl3RkV5WDlLSFVIU1NFIiwiYWdlIiwyNV0'
    }
  ];
 ```
 
and the `revealDisclosures` might be better named as `selectDisclosures` or `includeDisclosures`
as it is meant include only the selected disclosures in the SDJWT for the presentation of it.

We can make those changes after we accept this PR